sparse/usr/bin/droid/droid-hal-shutdown.sh

```diff
@@ -54,41 +54,42 @@ PREV_NUM_PIDS=$NUM_PIDS
 # We don't use it, but some init scripts watch for it as a signal to shut other things down
 /usr/bin/setprop sys.shutdown.requested 1
 
-echo Shutting down droid-hal-init services
+echo "Shutting down droid-hal-init services"
 /usr/bin/setprop hybris.shutdown 1
 
 sleep 1
 WAIT=1
 get_pids
 MAX_WAIT=5
 # -gt 1 because droid-hal-init is also in this cgroup
-while [ $NUM_PIDS -gt 1 -a $WAIT -lt $MAX_WAIT ]; do
+while [ $NUM_PIDS -gt 1 ] && [ $WAIT -lt $MAX_WAIT ]; do
     let WAIT=$WAIT+1
     if [ $NUM_PIDS -lt $PREV_NUM_PIDS ]; then
         # Number of running processes is getting smaller
         # Wait a little bit more
         sleep 1
      else
-        # Number of pids is not gettting smaller
+        # Number of pids is not getting smaller
         break
     fi
     PREV_NUM_PIDS=$NUM_PIDS
     get_pids
 done
 
-echo Killing droid-hal-init
+echo "Killing droid-hal-init"
 killall droid-hal-init
 
-echo Killing processes hybris.shutdown missed
 get_pids
 if [ $NUM_PIDS -gt 0 ]; then
-    killall $PIDS
+    echo "Terminating processes hybris.shutdown missed"
+    kill -TERM $PIDS
     sleep 1
     get_pids
     if [ $NUM_PIDS -gt 0 ]; then
-        killall -s 9 $PIDS
+        echo "Killing processes hybris.shutdown missed"
+        kill -KILL $PIDS
     fi
 fi
 
-exit 0
-
+get_pids
+exit 0
```

Comment:
You removed the new line at the end of the file here.

Reply:
Fixed.